ingestion_server/ingestion_server/es_mapping.py

```diff
@@ -101,6 +101,7 @@ def index_settings(table_name):
                 "fields": {"keyword": {"ignore_above": 256, "type": "keyword"}},
                 "type": "text",
             },
+            "filetype": {"type": "keyword"},
```

Comment:
Can we name this `extension` so that it's clear we're not actually extracting filetype information?

Reply:
The thing is that the extension(s) is still the compilation of filetypes so here, this will still represent the filetype of the preview or the single audio file. The Audio model is keeping the `filetype` field name, are we going to change that too? (question for @obulat and the rest of @WordPress/openverse-maintainers) I believe that deserves its own separate discussion because it would be better to be consistent between the Audio and Image models.

Reply:
I feel like we keep kicking this conversation down the road though and I'm not sure where to have it or how to come to a resolution on it. We have discussed it in a variety of places and come down to this completely quiet issue: https://github.com/WordPress/openverse-catalog/issues/478

I'm concerned that the way we're going now, of putting this information into ES the way we ware or doing it in the API or anything like that without first evaluating the consequences of it and how we want to ideally do it in the future, we'll lock ourselves into a bad situation and have our audio v1 API full of messy, incongruous concepts... I could be being overly cautious though. And I don't want to stall this work which will improve the existing non-ideal implementation.

I'm hesitant to just keep pushing these discussions down the road though and wonder what the best way to prioritize it because it keeps coming up over and over again.

Reply:
The way I see it "filetype" and "extension" are pretty much synonymous in our case. So perhaps changing the filter param name (from `extension` to `filetype`) would make the situation clearer? That name is from CC times. The other way seems that is much more work and I'm not sure if it's worth it.

> We have discussed it in a variety of places and come down to this completely quiet issue: https://github.com/WordPress/openverse-catalog/issues/478

That's unfortunate, that issue was off my radar so sadly I don't think it will enter the milestone. My vote would be to ditch that field at this point, we're not using it.

> I'm concerned that the way we're going now, of putting this information into ES the way we ware or doing it in the API or anything like that without first evaluating the consequences of it and how we want to ideally do it in the future, we'll lock ourselves into a bad situation and have our audio v1 API full of messy, incongruous concepts...

I have exactly the same thoughts when it comes to adding new media types. I don't exactly have the answer to this problem either, other than to prioritize at some point, and hope that new types of media are better planned with the lessons learned from audio.

Reply:
No, specifically the discussion related to the issue I linked, that is, metadata extraction from files during ingestion (or during some other stage in our data pipeline, whenever it may be). It is related to the waveform discussion as well as waveform data is a form of metadata and requires us to download the files to produce it.

I'm not advocating for needing to decide and implement this now. I'm just worried that if we don't at least have _some_ idea of how it might happen or indeed if it will happen at all, then we'll do things like mix together filetype and extension. If the project decided it didn't want to do this then at least we could just ignore the question entirely and do whatever is easiest for the moment. Right now, however, it's just hanging over these metadata decisions. Some of these decisions could lock us into APIs we wouldn't want in the future if we do decide to do something different with metadata.

Reply:
Actually, I thought for a moment to follow that path of grouping extensions too, as @obulat did something similar for [image extensions in the catalog](https://github.com/WordPress/openverse-catalog/pull/545) and I found that .ogg and .oga can be considered the same filetype but then filetype for images are limited to one media file, it's not the same case for audio.

Honestly, I'm not very familiar with the similarity between many filetypes/container formats, and from what I see there seems to be more variety of containers for audio formats. So I thought it may be actually better to have the finer grain filter, if someone wants the whole group it could simply pass all the extensions and it would be the same result.

I agree that the `bitrate` and the `waveform` will probably follow a similar approach, and I want to investigate it, but I believe `filetype` can be as simple as this, given we already have that data in the catalog (it's [confirmed](https://github.com/WordPress/openverse-catalog/issues/559#issuecomment-1164343209) that all the audio files have this field).